HBT.cpp

```diff
@@ -93,6 +94,8 @@ int main(int argc, char **argv)
 
     timer.Tick(world.Communicator);
     subsnap.AssignHosts(world, halosnap, partsnap);
+    MergerTreeInfo merger_tree;
```

Comment:
It might be cleaner to have the merger tree as a member of subsnap. We could then do StoreTracerIds within UpdateParticles, and then identify the descendants within UpdateTracks.

Reply:
I don't think I can do that easily. If I put a MergerTreeInfo instance in the subsnap the class has to be defined at that point which means including merger_tree.h in subhalo.h, but merger_tree.h uses types from subhalo.h so I get a circular dependency. I think I would need to either duplicate the forward declaration of SubhaloList_t from subhalo.h in merger_tree.h or just move the contents of merger_tree.h into subhalo.h.